Fetch application services failing and breaking CI

Looks like our script to fetch application services is failing and breaking CI.

Seen here: https://app.circleci.com/pipelines/github/mozilla/experimenter/35306/workflows/083c5e34-f131-4e56-aed1-a1455dcc4bd2/jobs/158256

```sh
2.258 Do you want to continue? [Y/n] Abort.
------
Dockerfile:6
--------------------
   5 |     
   6 | >>> RUN apt-get update && \
   7 | >>>     apt-get upgrade && \
   8 | >>>     apt-get install -y \
   9 | >>>         curl \
  10 | >>>         jq \
  11 | >>>         zip && \
  12 | >>>     ./fetch-application-services.sh
  13 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update &&     apt-get upgrade &&     apt-get install -y         curl         jq         zip &&     ./fetch-application-services.sh" did not complete successfully: exit code: 1
make: *** [Makefile:106: build_megazords] Error 1
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4368)

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -23,7 +23,7 @@ commands:
             echo "Changed files:"
             echo "$diff_output"
 
-            if echo "$diff_output" | grep -E '<< parameters.paths >>|^[^/]+$|^.circleci/'
+            if echo "$diff_output" | grep -E '<< parameters.paths >>|^[^/]+$|^.circleci/|application-services/'
               then
                 echo "Changes detected in << parameters.paths >> or .circleci or root directory. Running tests and linting."
               else

--- a/application-services/Dockerfile
+++ b/application-services/Dockerfile
@@ -1,12 +1,12 @@
-FROM debian:bullseye
+FROM debian:bullseye-20240130
 WORKDIR /application-services
 
 COPY . .
 
 RUN apt-get update && \
-    apt-get upgrade && \
+    apt-get upgrade -y && \
     apt-get install -y \
-        curl \
-        jq \
-        zip && \
+    curl \
+    jq \
+    zip && \
     ./fetch-application-services.sh

